apiserver/common/networkingcommon/networkconfigapi.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// The networkconfigapi package implements the network config parts
+// common to machiner and provisioner interface
+
+package networkingcommon
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+type NetworkConfigAPI struct {
+	st           *state.State
```

Comment:
Ideally this would take the minimal 'interface' that it actually needs, which is backed by a state.State object, rather than depending directly on state.

From what I see with a quick grep, the only thing we depend on is st.FindEntity so using

type BackingAPI interface {
  FindEntity(?) ...
}

Would work. That would also let you drop in a simple Mock for the test suite, and allow us to use something much lighter for test setup.